File: ObjectExternal/DemoCatalog.js
//-------------------------------------------------------
// Server side script for catalog API
//-------------------------------------------------------

DemoCatalog.display = function(params) {
	// ZZZ API external object ZZZ
	this.setJSONMIMEType();
	
	var method = params.getMethod();
	console.log("Method = " + method);
	
	var req = params.getJSONObject();
	console.log("Request = " + req);

	var ps = new JSONArray();
	var prd = this.getGrant().getTmpObject("DemoProduct");
	var rows = prd.search();
	for (var i = 0; i < rows.size(); i++) {
		prd.setValues(rows.get(i));
		var p = new JSONObject();
		p.put("ref", prd.getFieldValue("demoPrdReference"));
		p.put("name", prd.getFieldValue("demoPrdName"));
		ps.put(p);
	}
	
	var res = new JSONObject();
	if (req) res.put("request", req);
	res.put("response", ps);
	return res.toString();
};

File: src/com/simplicite/extobjects/Demo/DemoCatalog.java
package com.simplicite.extobjects.Demo;

import java.util.List;

import org.json.JSONArray;
import org.json.JSONObject;

import com.simplicite.objects.Demo.DemoProduct;
import com.simplicite.util.AppLog;
import com.simplicite.util.ExternalObject;
import com.simplicite.util.tools.Parameters;

/**
 * Catalog custom webservice
 */
public class DemoCatalog extends ExternalObject {
	private static final long serialVersionUID = 1L;

	/**
	 * Display method
	 * @param params Request parameters
	 */
	@Override
	public Object display(Parameters params) {
		// ZZZ API external object ZZZ
		setJSONMIMEType();

		String method = params.getMethod();
		AppLog.info(getClass(), "display", "Method = " + method, getGrant());

		JSONObject req = params.getJSONObject();
		AppLog.info(getClass(), "display", "Request = " + req.toString(), getGrant());

		JSONArray ps = new JSONArray();
		DemoProduct prd = (DemoProduct)getGrant().getTmpObject("DemoProduct");
		List<String[]> rows = prd.search();
		for (int i = 0; i < rows.size(); i++) {
			prd.setValues(rows.get(i));
			JSONObject p = new JSONObject();
			p.put("ref", prd.getFieldValue("demoPrdReference"));
			p.put("name", prd.getFieldValue("demoPrdName"));
			ps.put(p);
		}

		JSONObject res = new JSONObject();
		if (req!=null) res.put("request", req);
		res.put("response", ps);
		return res.toString();
	}
}
